apps/cowswap-frontend/src/modules/erc20Approve/containers/ApprovalAmountInput/ApprovalAmountInput.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { ReactNode, useCallback, useMemo } from 'react'
+
+import { TokenWithLogo } from '@cowprotocol/common-const'
+import { tryParseCurrencyAmount } from '@cowprotocol/common-utils'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { Field } from 'legacy/state/types'
+
+import { useUsdAmount } from 'modules/usdAmount'
+
+import * as styledEl from './styled'
+
+import { CurrencyInputPanel } from '../../../../common/pure/CurrencyInputPanel'
+import { CurrencyInfo } from '../../../../common/pure/CurrencyInputPanel/types'
+import { useCustomApproveAmountInputState, useUpdateOrResetCustomApproveAmountInputState } from '../../state'
+
+export function ApprovalAmountInput({
```

Comment:
I see somewhere you defined props as a separate interface, somewhere they are defined as function argument type. Let's be consistent and use interfaces everywhere